parcels/uxfieldset.py

```diff
@@ -0,0 +1,189 @@
+        n_nodes = self.uxds.uxgrid.n_nodes_per_face[fi].to_numpy()
+        node_ids = self.uxds.uxgrid.face_node_connectivity[fi, 0:n_nodes]
+        nodes = np.column_stack(
+            (
+                np.deg2rad(self.uxds.uxgrid.node_lon[node_ids].to_numpy()),
+                np.deg2rad(self.uxds.uxgrid.node_lat[node_ids].to_numpy()),
+            )
+        )
+
+        coord = np.deg2rad([x, y])
+        bcoord = np.asarray(_barycentric_coordinates(nodes, coord))
+        if ( not (bcoord >= 0).all() ) and (not (bcoord <= 1.0).all()):
+            return False
+        
+        return True
+
+    def eval(self, field_names, time, z, y, x, ei: int=None, applyConversion=True):
+        
+        res = {}
+
+        if ei is not None:
+            fi = ei
+            if not self._point_is_in_face(y,x,ei):
+                # If the point is not in the previously defined face, then
+                # search for the face again.
+                # To do : Update the search here to do nearest neighbors search, rather than spatial hash - joe@fluidnumerics.com
+                print(f"Position : {x}, {y}")
+                print(f"Hash indices : {self._spatialhash._hash_index2d(np.deg2rad([[x,y]]))}")
+                fi, bcoords = self._spatialhash.query([[x,y]]) # Get the face id for the particle  
+                fi = fi[0] 
+                print(f"Face index (updated): {fi}")
+                print(f"Barycentric coordinates (updated): {bcoords}")
+
+        for f in field_names:
+            field = getattr(self.uxds, f)
+            face_registered = ("n_face" in field.dims)
+
+            if face_registered:
+                r = self._face_interp(field, time, z, y, x, fi)
+            else:
+                r = self._node_interp(field, time, z, y, x, fi)
+
+            #if applyConversion:
+            #    res[f] = self.units.to_target(r, z, y, x)
+            #else:
+            # To do : Add call to units.to_target to handle unit conversion : joe@fluidnumerics.com
+            res[f] = r/111111.111111111
+            
+        return res, fi
+            
+def _barycentric_coordinates(nodes, point):
+    """
+    Compute the barycentric coordinates of a point P inside a convex polygon using area-based weights.
+    So that this method generalizes to n-sided polygons, we use the Waschpress points as the generalized
+    barycentric coordinates, which is only valid for convex polygons.
+
+    Parameters
+    ----------
+        nodes : numpy.ndarray
+            Spherical coordinates (lon,lat) of each corner node of a face
+        point : numpy.ndarray
+            Spherical coordinates (lon,lat) of the point
+    Returns
+    -------
+    numpy.ndarray
+        Barycentric coordinates corresponding to each vertex.
+
+    """
+    n = len(nodes)
+    sum_wi = 0
+    w = []
+
+    for i in range(0, n):
+        vim1 = nodes[i - 1]
+        vi = nodes[i]
+        vi1 = nodes[(i + 1) % n]
+        a0 = _triangle_area(vim1, vi, vi1)
```

Comment:
These areas need to be clipped with a minimum value. As we found in uxarray, when a particle lies on a face edge or vertex, these areas can be zero, leading to division by zero.